server.js

```diff
@@ -38,6 +40,43 @@ fileModule.walkSync('./endpoints', function iterateEndpoints(dirPath, dirs, endp
     }
 });
 
+app.use(function(error, req, res, next){
+  var code = 500;
+  var message = 'Unknown error';
+
+  if (typeof error === 'number') {
+    //next(404)
+    code = error
+    message = statuses[code] || message;
+  }else if (typeof error === 'object' && error.message && error.message.length === 3 && !isNaN(error.message)) {
+    //throw Error(404)
+    code = parseInt(error.message,10);
+    message = statuses[code] || message;
+
+  }else if(typeof error === 'object' && !error.code){
+    //throw Error('foo') or throw Error('404|foo')
```

Comment:
Is this `404|foo` thing used anywere in our codebase? I've never seen it before. Is it common?
What I have seen elsewhere is something like the following (but that's not covered by this error handler)

```
var err = new Error('foo');
err.status = 400;
return next(err);
```